### PR TITLE
[FIX] base: fix update translations

### PR DESCRIPTION
**Pull request by HydrionBurst**
for model_terms translations, update_field_translations uses {lang: {old_lang_term, new_lang_term}} to update translations This api is not designed for
1. one en_term to be translated to multiple different lang_terms
2. multiple en_terms to be translated to the same lang_term When the above corner cases happen, it would be impossible to rollback without the help of the `write` api.

For example
when the translations become
```
{
    "en_US": "<div> en1 </div><div> en2 <div>"
    "fr_FR": "<div> fr1 </div><div> fr1 <div>"
}
```
there is no way to make the two terms in fr_FR become different again. Unless the user `write` the value `with_context(lang="fr_FR")`.

It is acceptable for most fields as a workaround, but not ir.ui.view.arch_db, which we only display its logical en_US value to the user. There is no UI to directly modify the value with specified language.

This commit changes the api of `update_field_translations` and uses {lang: {old_en_term, new_lang_term}} to update translations. In the above example, the user can just call
`update_field_translations(field_name, {"fr_FR": {"en2": "fr2"}})`

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/fields.py [hunk 1]
```diff
@@ -1766,21 +1766,38 @@ def convert_to_record(self, value, record):
...
             base_lang = record._get_base_lang()
-            if base_lang != (record.env.lang or 'en_US'):
+            lang = record.env.lang or 'en_US'
+
+            if lang != base_lang:
                 base_value = record.with_context(edit_translations=None, check_translations=True, lang=base_lang)[self.name]
-                base_terms = self.get_trans_terms(base_value)
-                term_to_state = {term: "translated" if base_term != term else "to_translate" for term, base_term in zip(terms, base_terms)}
+                base_terms_iter = iter(self.get_trans_terms(base_value))
+                def get_base(term): return next(base_terms_iter)
```
Comment (kmagusiak) [inline odoo/fields.py:1777]: Use a lambda expression or format the code.

**Review by kmagusiak — APPROVED** [review]

**Review by rco-odoo — APPROVED** [review]
LGTM

**Comment by rco-odoo** [comment]
@robodoo override=ci/security

This code refactors existing code using the same function `sha256`.  Moreover, three out of five cases are for testing purpose.

**Comment by rco-odoo** [comment]
@robodoo override=ci/style

Assigning lambdas is actually fine.

**Comment by rco-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**